src/combobox/combobox.stories.ts

```diff
@@ -4,10 +4,58 @@ import { withKnobs, text, boolean } from "@storybook/addon-knobs/angular";
 
 import { ComboBoxModule } from "./combobox.module";
 import { DocumentationModule } from "./../documentation-component/documentation.module";
+import { Component, Input } from "@angular/core";
+
+@Component({
+	selector: "app-dynamic-combobox",
+	template: `
+		<ibm-combo-box
+			[disabled]="disabled"
+			[invalid]="invalid"
+			[invalidText]="invalidText"
+			[label]="label"
+			[helperText]="helperText"
+			[items]="items"
+			(search)="onSearch($event)">
+			<ibm-dropdown-list></ibm-dropdown-list>
+		</ibm-combo-box>
+	`
+})
+class ComboBoxStory {
+	@Input() disabled = false;
+	@Input() invalid = false;
+	@Input() invalidText = "Entry not valid";
+	@Input() label = "Combobox label";
+	@Input() helperText = "Optional helper text";
+	@Input() items = [
+		{
+			content: "one"
+		},
+		{
+			content: "two"
+		},
+		{
+			content: "three"
+		},
+		{
+			content: "four"
+		}
+	];
+
+	onSearch(event) {
+		// Set invalid to true if no search hits.
+		if (!this.items.filter(item => item.content.toLowerCase().includes(event.toLowerCase())).length) {
+			this.invalid = true;
+		} else {
+			this.invalid = false;
+		}
+	}
+}
```

Comment:
This could actually all be in the story ... `onSeach` would just need to be a `function` so it can access `this`

Reply:
Unless I'm misunderstanding, I can't access the props in the storybook through this in order to change the state of invalid.

Reply:
If you use a normal `function() {}` declaration instead of an arrow `() => {}` function, it gives you access to `this`, where `this` is the component that storybook creates!